src/utils/MainApi.js

```diff
@@ -0,0 +1,101 @@
+class MainApi {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Main API Error Type: 
+${res.status} ${res.statusText}`);
+    }
+  }
+
+  ////////=============== AUTH ================= //////////
+
+  async signup({ email, password, name }) {
+    console.log(email, password, name);
+    const response = await fetch(`${this._baseUrl}/signup`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({ email: email, password: password, name: name }),
+    });
+    return this._checkResponse(response);
+  }
+
+  signin({ email, password }) {
+    console.log("signin data:", email, password);
+    return fetch(`${this._baseUrl}/signin`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Accept: "application/json",
+      },
+      body: JSON.stringify({ email: email, password: password }),
+    })
+      .then(this._checkResponse)
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request`. And you don’t need to change anything else.